src/xmipp/applications/scripts/tomogram_reconstruction/tomogram_reconstruction.py

```diff
@@ -280,107 +334,124 @@ def astraReconstruction(self, ts_orig, tiltAngles, recAlgorithm):
-                                               verbose=True, blocksize=self.blocksize)
+                                               verbose=True, blocksize=self.blocksize,  noneg=False)
                        
-        elif self.method == 'awasd-pocs':
+        elif self.method == 'awasdpocs':
             epsilon = im3DNORM(tigre.Ax(algs.fdk(ts, geo, tiltAngles), geo, tiltAngles) - ts, 2) * 0.15
             reconstruction = algs.awasd_pocs(ts, geo, tiltAngles, self.iterations, tviter=self.tviter, maxl2err=epsilon, 
                                              alpha=self.alpha, lmbda=self.lmbda, lmbda_red=self.lambdared, rmax=self.ratio, verbose=True, 
-                                             delta=np.array([-0.005]))
+                                             delta=np.array([-0.005]), noneg=False)
             
-        elif self.method == 'irn-tv-cgls':
-            reconstruction = algs.irn_tv_cgls(ts, geo, tiltAngles, self.iterations, lmbda=self.lmbda, niter_outer=self.niter_outer)
+        elif self.method == 'irntvcgls':
+            reconstruction = algs.irn_tv_cgls(ts, geo, tiltAngles, self.iterations, lmbda=self.lmbda, niter_outer=self.niter_outer, noneg=False)
         
         elif self.method == 'fista':
-            reconstruction = algs.fista(ts, geo, tiltAngles, self.iterations, tviter=self.tviter, tvlambda=self.tvlambda)
+            reconstruction = algs.fista(ts, geo, tiltAngles, self.iterations, tviter=self.tviter, tvlambda=self.tvlambda, noneg=False)
             
-        elif self.method == 'sart-tv':
-            reconstruction = algs.sart_tv(ts, geo, self.alpha, self.iterations, self.tvlambda, self.tviter)
+        elif self.method == 'sarttv':
+            reconstruction = algs.sart_tv(ts, geo, tiltAngles, self.iterations, tvlambda=self.tvlambda, tviter=self.tviter, noneg=False)
             
         elif self.method == 'mlem':
-            reconstruction = algs.mlem(ts, geo, tiltAngles, self.iterations)
+            reconstruction = algs.mlem(ts, geo, tiltAngles, self.iterations, noneg=False)
         
         elif self.method == 'pcsd':
-            reconstruction = algs.pscd(ts, geo, tiltAngles, self.iterations)
+            reconstruction = algs.pscd(ts, geo, tiltAngles, self.iterations, noneg=False)
             
         elif self.method == 'awpcsd':
-            reconstruction = algs.awpscd(ts, geo, tiltAngles, self.iterations)
+            reconstruction = algs.awpscd(ts, geo, tiltAngles, self.iterations, noneg=False)
             
         else:
             raise  Exception('The selected reconstruction method does not exist, check the --method flag')
 
 
         with mrcfile.new(self.fnOut, overwrite=True) as mrc:
-            mrc.set_data(np.swapaxes(reconstruction, 0, 2))
+            mrc.set_data(np.transpose(reconstruction, (2, 0, 1)))
         
 
     def getMethod(self):
```

Comment:
Is this function used somewhere?

Reply:
In xmipp, it is not. Neither, in this code. This function is only used here